src/configuration.cpp

```diff
@@ -1,104 +1,109 @@
-                        }
-
-                        if(plugin_spec_cfg.find("minimum_delay_time") != plugin_spec_cfg.end()) {
-                            default_data_movement_parameters = plugin_spec_cfg.at("minimum_delay_time").get<std::string>();
-                        }
-
-                        if(plugin_spec_cfg.find("maximum_delay_time") != plugin_spec_cfg.end()) {
-                            default_data_movement_parameters = plugin_spec_cfg.at("maximum_delay_time").get<std::string>();
-                        }
-
-                        if(plugin_spec_cfg.find("time_check_string") != plugin_spec_cfg.end()) {
-                            time_check_string = plugin_spec_cfg.at("time_check_string").get<std::string>();
-                        }
-
-                        if(plugin_spec_cfg.find("number_of_scheduling_threads") != plugin_spec_cfg.end()) {
-                            number_of_scheduling_threads = plugin_spec_cfg.at("number_of_scheduling_threads").get<int>();
-                        }
-
-                        if(plugin_spec_cfg.find(data_transfer_log_level_key) != plugin_spec_cfg.end()) {
-                            const std::string& val = plugin_spec_cfg.at(data_transfer_log_level_key).get_ref<const std::string&>();
-                            if("LOG_NOTICE" == val) {
-                                data_transfer_log_level_value = LOG_NOTICE;
-                            }
-                        }
-                    }
-
-                    success_flag = true;
-                } // if inst_name
-            } // for rule_engines
-        } catch ( const boost::bad_any_cast& e ) {
-            THROW( INVALID_ANY_CAST, e.what() );
-        } catch ( const std::out_of_range& e ) {
-            THROW( KEY_NOT_FOUND, e.what() );
-        }
-
-        if(!success_flag) {
-            THROW(
-                SYS_INVALID_INPUT_PARAM,
-                boost::format("failed to find configuration for storage_tiering plugin [%s]") %
-                _instance_name);
-        }
-    } // ctor storage_tiering_configuration
-
-} // namepsace irods
+#include <irods/irods_server_properties.hpp>
 
+#include <fmt/format.h>
+
+namespace irods
+{
+	storage_tiering_configuration::storage_tiering_configuration(const std::string& _instance_name)
+		: instance_name{_instance_name}
+	{
+		bool success_flag = false;
+		try {
+			const auto& rule_engines = get_server_property<const nlohmann::json&>(
+				std::vector<std::string>{KW_CFG_PLUGIN_CONFIGURATION, KW_CFG_PLUGIN_TYPE_RULE_ENGINE});
+
+			for (const auto& rule_engine : rule_engines) {
+				if (const auto& inst_name = rule_engine.at(KW_CFG_INSTANCE_NAME).get_ref<const std::string&>();
+				    inst_name != _instance_name)
+				{
+					continue;
+				}
+
+				if (rule_engine.count(KW_CFG_PLUGIN_SPECIFIC_CONFIGURATION) <= 0) {
+					success_flag = true;
+					continue;
+				}
+
+				const auto& config = rule_engine.at(KW_CFG_PLUGIN_SPECIFIC_CONFIGURATION);
```

Comment:
This can be replaced with a `.find()`.

Reply:
True. What should we do if no plugin-specific configuration stanza is defined? Log a message and bail (or continue to the next REP)? Hmm...

Reply:
We have decided that we can treat the scenario where no `plugin_specific_configuration` is defined as using a default configuration with no errors.

Reply:
Done